src/core/components/layout/components/columns/styles.ts

```diff
@@ -135,23 +137,84 @@ const generateWidthCSS = (width: number | number[]) => {
 	return calculateWidth(width);
 };
 
-export const column = (width?: number | number[]) => {
-	let flex;
-	let widthCSS;
-
-	if (width == null || (Array.isArray(width) && width.length === 0)) {
-		// If no width is specified, allow the column to grow
-		flex = 1;
-		widthCSS = css``;
-	} else {
-		// If a width is specified, don't allow column to grow. Use the width property
-		flex = '0 0 auto';
-		widthCSS = generateWidthCSS(width);
+const generateSpanCSS = (span: number | number[]) => {
+	if (Array.isArray(span)) {
+		const breakpoints: Breakpoint[] = [
+			'mobile',
+			'tablet',
+			'desktop',
+			'leftCol',
+			'wide',
+		];
+
+		return span.reduce((styles, w, i) => {
+			return css`
+				${styles}
+				${from[breakpoints[i]]} {
+					${calculateSpan(w)};
+				}
+			`;
+		}, css``);
 	}
 
+	return calculateSpan(span);
+};
+
+const columnBreakpoints: Array<ColumnBreakpoint> = [
+	{ totalColumns: 4, rule: until.tablet },
+	{ totalColumns: 12, rule: between.tablet.and.leftCol },
+	{ totalColumns: 14, rule: between.leftCol.and.wide },
+	{ totalColumns: 16, rule: from.wide },
+];
+
+const calculateSpan = (span: number) => {
+	const columnBreakpointCss = columnBreakpoints.reduce(
+		(acc, cur: ColumnBreakpoint) => {
+			if (span === 0)
+				return css`
+					display: none;
+				`;
+
+			const inferredWidth = span / cur.totalColumns;
+			const cappedWidth = inferredWidth < 1 ? inferredWidth : 1;
+			const cssForBreakpoint = calculateWidth(cappedWidth);
+
+			return css`
+				${acc}
+				${cur.rule} {
+					${cssForBreakpoint}
+					display: block;
+				}
+			`;
+		},
+		css``,
+	);
+
 	return css`
 		box-sizing: border-box;
-		flex: ${flex};
-		${widthCSS};
+		flex: ${span ? '0 0 auto' : 1};
+		${columnBreakpointCss}
 	`;
 };
+
+// always gets applied
+export const column = css`
+	box-sizing: border-box;
+`;
+
+// get applied when no width or span is specified
+export const flexGrow = css`
+	flex: 1;
+`;
+
+// width is specified
+export const setWidth = (value: number | number[]) => css`
+	flex: 0 0 auto;
```

Comment:
I know you didn't introduce this, but can it be optimised to `flex: 0`?

Reply:
`flex: 0` caused some problems but `flex: auto` seemed to work - though I'd be a bit reluctant to change it in this PR without doing some proper testing around it.

This did make me realise that the `flex` property in setSpan below is redundant.